healthx/handler_test.go

```diff
@@ -36,8 +36,12 @@ import (
 	"github.com/ory/herodot"
 )
 
-func TestHealth(t *testing.T) {
+const mockHeaderKey = "middleware-header"
+const mockHeaderValue = "test-header-value"
+
+func coreHealthTests(t *testing.T, withMiddleware bool) {
```

Comment:
Just a bit of a nit
I just wanted these lines https://github.com/ory/x/pull/514/files#diff-519494e59fcbc1e8e99737385005a0c4b843639063e9ad54f4a22d3ba3baf646R98-R129 abstracted then you will always just call like so instead of doing some if statement logic which makes it more difficult to read in future.

```go
t.Run("case:without middleware", func(t *testing.T) {
    // do handler stuff
    assertDefaultResponse(t, false)
})

t.Run("case:with middleware", func(t *testing.T) {
    // do middleware stuff with health and version
    assertDefaultResponse(t)
    // assert middleware response
})
```

Reply:
Done, 
Though, maybe the approach I've used is not what you were thinking. 

Also, a tiny doubt 
https://github.com/ory/x/pull/514/commits/5ffdf2020e90ed3de88338572c859e9615c7c7a7#diff-519494e59fcbc1e8e99737385005a0c4b843639063e9ad54f4a22d3ba3baf646R46-R48

I'm modifying the handler internally since I didn't wanna pass the `error` typed variable around, unspecific to the test, doesn't that mean I'm modifying behaviour inside the abstraction and that's not right?